File: swift/problem198.swift
//nearly the same as problem746
class Solution {
    func rob(_ nums: [Int]) -> Int {
        var max1 = 0; var max2 = 0; var cur = 0
        
        for cash in nums {
            cur = max(cash+max2,max1)
            max2 = max1
            max1 = cur
        }
        
        return max(max1,max2)
    }
}

File: python/problem198.py
class Solution:
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        if len(nums)<3:
            return max(nums)
        li = []
        li.append(nums[0])
        li.append(nums[1])
        li.append(nums[0] + nums[2])
        
        for i in range(3,len(nums)):
            li.append(nums[i]+max(li[i-2],li[i-3]))
        return max(li[-2],li[-1])
        